src/lang/state.clj

```diff
@@ -0,0 +1,47 @@
+(ns lang.state
+  (:require [clojure.core.cache.wrapped :as cache]
+            [taoensso.timbre :as log]))
+
+(def state (atom {}))
+
+(defn void! []
+  (reset! state {}))
+
+(defn get! [path]
+  (reduce
+   (fn [val step]
+     (if (delay? val)
```

Comment:
Far from the most important, but was just skimming -- IIRC `force` works on all values, delay or not, so it's handy for just throwing on any maybe-delay expression.